src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -874,13 +874,14 @@ public boolean isInterrupted() {
     @Override
     boolean getAndClearInterrupt() {
         assert Thread.currentThread() == this;
-        synchronized (interruptLock) {
-            boolean oldValue = interrupted;
-            if (oldValue)
+        boolean oldValue = interrupted;
+        if (oldValue) {
+            synchronized (interruptLock) {
```

Comment:
Don't you still need to read under the lock to synchronize with the interrupt method?

Reply:
The override of interrupt for virtual threads always sets the interrupt status as the first step. So if Thread.interrupted reads the interrupt status as true then it will synchronize. It's not too different to platform threads, I think you added a comment to Thread.interrupted about this when move the interrupted flag into Java.